src/validation/nationality.validation.ts

```diff
@@ -0,0 +1,142 @@
+          return maxLengthNationalityValidationResult;
+      }
+  }
+  return undefined;
+}
+
+export const validateInvalidCharacterValuesForNationality = (nationality: string[], nationalityValidationType: NationalityValidationType): ValidationError | undefined => {
+  if(nationality[0] && (!nationality[0].trim().length || invalidPattern(nationality[0], VALID_NATIONALITY_CHARACTER))) {
+      return nationalityValidationType.Nationality1InvalidCharacter.Nationality;
+  }
+  if(nationality[1] && (!nationality[1].trim().length || invalidPattern(nationality[1], VALID_NATIONALITY_CHARACTER))) {
+    return nationalityValidationType.Nationality2InvalidCharacter.Nationality;
+  }
+  if(nationality[2] && (!nationality[2].trim().length || invalidPattern(nationality[2], VALID_NATIONALITY_CHARACTER))) {
+    return nationalityValidationType.Nationality3InvalidCharacter.Nationality;
+  }
+  return undefined;
+}
+
+export const validateInvalidLengthValuesForNationality = (nationality: string[], nationalityValidationType: NationalityValidationType): ValidationError | undefined => { 
+  if(nationality[0] && (nationality[0].length > 50)) {
+      return nationalityValidationType.Nationality1LengthValidator.Nationality
+  }
+  if(nationality[1] && (nationality[1].length > 50)) {
+    return nationalityValidationType.Nationality2LengthValidator.Nationality
+  }
+  if(nationality[2] && (nationality[2].length > 50)) {
+    return nationalityValidationType.Nationality3LengthValidator.Nationality
+  } 
+  return undefined;
+}
+
+export const validateAllowedListNationality = (nationality: string[], nationalityValidationType: NationalityValidationType): ValidationError | undefined => {
+  const nationalityList = NATIONALITY_LIST.split(";");
+  if(nationality[0] && !nationalityList.includes(nationality[0])) {
+    return nationalityValidationType.Nationality1AllowedList.Nationality
+  }  
+  if(nationality[1] && !nationalityList.includes(nationality[1])) {
+    return nationalityValidationType.Nationality2AllowedList.Nationality
+  }
+  if(nationality[2] && !nationalityList.includes(nationality[2])) {
+  return nationalityValidationType.Nationality3AllowedList.Nationality
+  } 
+return undefined;
+} 
+
+export const validateDuplicateNationality = (nationality: string[], nationaliltyValidationType: NationalityValidationType) => {
+  if((nationality[0] && nationality[1]) && (nationality[1] ===  nationality[0])) {
+    return nationaliltyValidationType.DuplicatedNationality2Validator.Nationality
+  }
+  if((nationality[0] && nationality[1] && nationality[2]) && (nationality[2] ===  nationality[0] || nationality[2] === nationality[1])) {
+    return nationaliltyValidationType.DuplicatedNationality3Validator.Nationality
+  }
+}
+
+export const validateMaximumLengthNationality = (nationality: string[], nationaliltyValidationType: NationalityValidationType) => {
+  if ((nationality) && (`${nationality[0]},${nationality[1]},${nationality[2]}`.length > 48)) {
```

Comment:
In the API we allow 49 characters for 2 nationalities.

Reply:
Dual nationality character added